core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1091,10 +1091,6 @@ public boolean isCompatible(String subject,
 
     List<ParsedSchema> prevParsedSchemas = new ArrayList<>(previousSchemas.size());
     for (Schema previousSchema : previousSchemas) {
-      if (previousSchema == null) {
-        throw new InvalidSchemaException(
-            "Existing schema " + previousSchema + " is not a valid Avro schema");
-      }
```

Comment:
Why did we remove this ?

Reply:
This is obsolete code.  The input schemas are never null.   Previously when parsing a schema failed, we returned a null.  Now when parsing a schema fails, we throw an exception in the `parseSchema` method.

Reply:
Got it !